src/assemble_workflow/fpm_builder.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import shutil
+import subprocess
+from assemble_workflow.bundle_recorder import BundleRecorder
+
+
+class FpmBuilder:
+    def build(self, bundle_recorder: BundleRecorder, archive_path: str, distribution: str) -> None:
+        manifest_data = bundle_recorder.bundle_manifest.data["build"]
+        product_name = product_name_alt = manifest_data["name"].lower()
+        if product_name == "opensearch dashboards":
+            product_name = "-".join(product_name.split())
+            product_name_alt = "_".join(product_name.split("-"))
+        root = os.path.abspath(os.path.join(__file__, os.pardir, os.pardir, os.pardir))
+        scripts_path = os.path.join(root, "scripts", "pkg")
+        architecture_alt = "x86_64" if manifest_data["architecture"] == "x64" else "aarch64"
+        real_archive_path = os.path.realpath(archive_path)
+        data_dir_path = os.path.join(real_archive_path, "data", "")
+        log_dir_path = os.path.join(real_archive_path, "logs", "")
+        systemed_entrypoint_path = os.path.join(real_archive_path, "bin", "systemd-entrypoint")
+        os.makedirs(data_dir_path, exist_ok=True)
+        os.makedirs(log_dir_path, exist_ok=True)
+        shutil.copyfile("scripts/pkg/scripts/systemd-entrypoint", systemed_entrypoint_path)
+        os.chmod(systemed_entrypoint_path, 0o755)
+        subprocess.run(
```

Comment:
This ignores errors, needs to be something like `check_run`.

Reply:
You mean `check_call`?
I think I can also use `execute`? Since that one internally called `check_returncode`.

Reply:
Anyways, use check_call as of now.
Thanks.